tests/snappi_tests/intf_utils/intf_accept_metrics.py

```diff
@@ -0,0 +1,109 @@
+#               +-----+   |                     |
+#               |  .  |   | +-----+ +-----+     |
+#               |  .  |   | |value| |flags|     |
+#               |  .  |   | +-----+ +-----+     |
+#               |  .  |   +---------------------+
+#               |  .  |              .
+#               |  .  |              .
+#               |  .  |              .
+#               |  .  |   +---------------------+
+#               |  .  |   | DataPoint M         |
+#               +-----+   | +---------+ +-----+ |
+#               |  M  |-->| |timestamp| |label| |
+#               +-----+   | +---------+ +-----+ |
+#                         |                     |
+#                         | +-----+ +-----+     |
+#                         | |value| |flags|     |
+#                         | +-----+ +-----+     |
+#                         +---------------------+
+
+from typing import List, Dict, Union
+
+class NumberDataPoint:
+    def __init__(self, time_unix_nano: int, label: List[Dict[str, str]], value: Union[int, float], flags: int = None):
+        self.time_unix_nano = time_unix_nano    # UNIX Epoch time in nanoseconds
+        self.label          = label             # The key of key-value pairs in dictionaries
+        self.value          = value             # Metric value (can be double or integer)
+        self.flags          = flags             # Optional flags
+
+    def __repr__(self):
+        return (f"NumberDataPoint(label={self.label}, "
+                f"time_unix_nano={self.time_unix_nano}, value={self.value}, flags={self.flags})")
+
+
+class Gauge:
+    def __init__(self):
+        self.data_points = []  # List of NumberDataPoint objects
+
+    def add_data_point(self, data_point):
+        self.data_points.append(data_point)
+
+    def __repr__(self):
+        return f"Gauge(data_points={self.data_points})"
+
+
+class Metric:
+    def __init__(self, name, description, unit, data_points, metadata=None):
+        self.name        = name            # Metric name
+        self.description = description     # Metric description
+        self.unit        = unit            # Metric unit (e.g., seconds, bytes)
+        self.data        = data            # Can be Gauge only
+        self.metadata    = metadata or {}  # Default to an empty dictionary if None
+
+    def __repr__(self):
+        return (f"Metric(name={self.name}, description={self.description}, "
+                f"unit={self.unit}, data={self.data})")
+
+
+# a ScopeMetrics object's ID is device_id
+class ScopeMetrics:
+    def __init__(self, device_id):
+        self.device_id = device_id
+        self.metrics   = []
+
+    def add_metric(self, metric):
+        self.metrics.append(metric)
+
+    def __repr__(self):
+        return f"ScopeMetrics(scope={self.scope}, metrics={self.metrics})"
+
+
+# a ResourceMetrics object's ID is test_run_id
+class ResourceMetrics:
+    def __init__(self, test_run_id, os_version):
```

Comment:
I believe we have way more than these 2 fields, do you mind to help check our meeting notes and add them in?

Reply:
Added testbed_name, testcase_name.